crates/terminal/src/terminal.rs

```diff
@@ -425,7 +425,7 @@ impl TerminalBuilder {
         // Optional suffix matches MSBuild diagnostic suffixes for path parsing in PathLikeWithPosition
         // https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-diagnostic-format-for-tasks
         let word_regex =
-            RegexSearch::new(r#"[\$\+\w.\[\]:/\\@\-~]+(?:\((?:\d+|\d+,\d+)\))?"#).unwrap();
+            RegexSearch::new(r#"[\$\+\w.\[\]:/\\@\-~()]+(?:\((?:\d+|\d+,\d+)\))?"#).unwrap();
```

Comment:
Don't we need to escape these so they're not treated as a capture group?

```suggestion
            RegexSearch::new(r#"[\$\+\w.\[\]:/\\@\-~\(\)]+(?:\((?:\d+|\d+,\d+)\))?"#).unwrap();
```

Reply:
We don't need to escape parentheses **inside a character class** because parentheses in this context are treated as literal characters, not as special regex constructs like capture groups.

I was able to repro the issue and the confirm that this change fixed it.
<img width="1432" alt="Screenshot 2024-09-07 at 7 55 32 AM" src="https://github.com/user-attachments/assets/5c147af0-990e-4cc7-b222-75823d383f04">